### ISSUE TITLE
Migrate JavaFX examples or retire them?

### ISSUE THREAD

**Issue by keilw** [JavaMoney/javamoney-examples#29]
As we heard, **Java 11** is not going to include JavaFX any more. 

The JavaFX examples either have to be migrated to a successor project (Open FX or whatever) or retired over time if JavaFX itelf also ends up being archived.

**Comment by msgilligan** [comment]
I'm hearing that Java FX being decoupled from Java will be a good thing -- that it will evolve faster. Won't it just be an optional Maven dependency? By "migrate" do you just mean add the optional dependency?

**Comment by msgilligan** [comment]
> Well it won't work with the JDK any more after Java 11

won't work with **only** the JDK any more after Java 11, right?

**Comment by aalmiray** [comment]
@msgilligan correct. OracleJDK 11 will ship without JavaFX. This is the same case as OpenJDK does right now, you must download OpenJFX as a separate binary. Starting with Java 11 we'll see releases of JavaFX that may or may not track the same release dates as OpenJDK; we're working on it.

**Comment by keilw** [comment]
@aalmiray Thanks for the input. Has the new place, namespace (e.g. if "javafx" may be preserved at least for existing modules like JakartaEE does) etc. been found already?

**Comment by keilw** [comment]
Marked it external because there is a dependency on a TBD JavaFX project.

**Comment by aalmiray** [comment]
No migration to a different space/brand is needed at the moment. JavaFX will continue being developed as part of the OpenJDK umbrella under OpenJFX.

**Comment by keilw** [comment]
If the community is allowed to do that like say DeviceIO (not sure, if that's still alive?) then it sounds like a good idea. Any chance you might get the RI of JSR 377 into that optional OJDK project or is Griffon the better option?

**Comment by aalmiray** [comment]
I think releasing an RI for JSR377 would be faster if Griffon is retrofitted with the new API rather than starting from scratch 😉

**Comment by keilw** [comment]
As long as the EC still exists and extends the lifeline (I see no Renewal Ballot 3 scheduled, but 13 months after the last one it should have been due by now) why not.

**Comment by msgilligan** [comment]
As far as I'm concerned I'd be happy to see the examples support OpenJFX 11 and later only.

**Comment by keilw** [comment]
We made this switch in the master/2.0 branch for UoM. I can't say, if the 1.x version of the Money JSR should already switch entirely or if we should have an OpenJFX 11 and JDK JavaFX 8 compatible version side-by side? Running in CI only makes sense for one. I am not opposed to keeping a javafx-java8 kind of module but disable it and use OpenJDK 11 to build the demos. Nothing keeps those who need the Java 8 compliant ones to run them, the minimum Java version would only be 11 for this one.

WDYT?

**Comment by Daniel-Dos** [comment]
Hi,

I can leave the two versions, however with java 11 disabled, as was done for the example javamoney-console-java10 .

**Comment by keilw** [comment]
Guess that (supporting JFX for Java 8 and 11+) might require profiles and the toolchain plugin, something that already works in other places, but is a bit complicated. I guess it is best to upgrade the JFX modules and use a minimum JDK like 11.

**Comment by keilw** [comment]
@aalmiray Btw what happened to 377?
It looks pretty dead, hasn't had a Renewal Ballot since 2 years, which is unacceptable for a JSR under JCP 2.11, but it still claims to be "active", will it be withdrawn or what's the plan?

**Comment by keilw** [comment]
Migrating them to OpenJFX 11 for 1.3 examples. Future versions should grow with the minimum JDK version
